lib/whitehall_importer/integrity_checker/change_history_check.rb

```diff
@@ -2,28 +2,32 @@ module WhitehallImporter
   class IntegrityChecker::ChangeHistoryCheck
     attr_reader :edition, :proposed_change_history, :publishing_api_change_history
 
-    def initialize(proposed_change_history, publishing_api_change_history, edition)
-      @proposed_change_history = proposed_change_history
-      @publishing_api_change_history = publishing_api_change_history
+    def initialize(edition, proposed_change_history, publishing_api_change_history)
       @edition = edition
+      @proposed_change_history = trim_proposed_change_history(proposed_change_history)
+      @publishing_api_change_history = publishing_api_change_history
     end
 
     def match?
-      if publishing_api_change_history.empty?
+      if publishing_api_change_history.empty? && edition.live?
         return proposed_history_has_first_published_change_note?
       end
 
       return false unless history_length_matches?
 
-      if edition.live?
-        history_matches?(proposed_change_history, publishing_api_change_history)
-      else
-        history_excluding_first_timestamp_matches?
-      end
+      history_matches?(proposed_change_history, publishing_api_change_history)
     end
 
   private
 
+    def trim_proposed_change_history(proposed_change_history)
+      if edition.major? && edition.change_note && !edition.live? || !edition.live? && edition.first?
```

Comment:
It maybe just me, and you guys may not agree... Anyway, just a suggestion, but I'd prefer some brackets in this - tends to make the logic more obvious and that whole thing easier to read and parse, e.g.

```
if ((edition.major? && edition.change_note && !edition.live?) || (!edition.live? && edition.first?))
```

Reply:
Good point about it being a bit hard to parse, I don't normally use brackets as it doesn't make it any easier for me personally to read. I have taken your point on board though and created variables to better describe the conditional arguments which I think it makes it much clearer.